### ISSUE TITLE
Reason for updating reward for consecutive frames

### ISSUE THREAD

**Issue by dandelionhsc** [TonyLianLong/VAI-ReinforcementLearning#3]
The function 'update_inv_and_rewards' in agent.py is different from the 'PAD'. Can you explain why you put the part 'update rewards'? I think this part is trying to match the reward between observation and next observation, am I understanding correctly? Also I couldn't find this part in the paper, is it explained in the paper?
Thank you very much.

**Comment by TonyLianLong** [comment]
The main idea is: the reward prediction is used for a more generalized task which we did not include in VAI. You could use the predicted rewards if you extend VAI to the general framework, but removing the reward prediction part generally has no effect.

More details:
We originally attempted to propose a more general framework that later came to be VAI. In the general framework, not only do the observation change (i.e. distractions) from training to adaptation (deployment), but effects of action also change. Think of this as an example: a robot running out of battery gives a lighter force compared to a robot with battery charged. Through predicting the reward from observations, we will be able to adapt our agent during deployment with our pseudo-reward predicted from observations in the consecutive frames: we want our agent's behavior to be constant, no matter the "battery status". This is working well in many cases: if you multiply a scalar (e.g. 0.5 to 0.8) to all actions in Walker, the agent is still able to adapt with the pseudoreward.

However, the problem is that adaptation during deployment, either with PAD or the method I mentioned, ask us to backprop in deployment, which is costly in terms of computation. In the end, we did not focus on the generalized case with different effect from the action between training vs. deployment, and we focus on the vision task: our main takeaway is to adapt vision instead of the agent, so to make it more coherent, we do not include the generalized part. You could use the predicted reward if you still want to generalize to task above (you need to modify the evaluation script to use predicted reward). The reason for including this part is to maintain compatibility with our checkpoints. If you don't use the predicted reward, it generally has no effect on the VAI with vision adaptation only (as we do not perform adaptation on actions in deployment and the predicted reward is not used).
